dali/benchmark/dali_bench.cc

```diff
@@ -22,6 +22,14 @@
 #include "dali/operators/operators.h"
 #include "dali/pipeline/operator/op_spec.h"
 
+// add this alignment to work around a patchelf bug/feature which
+// changes TLS alignment and break DALI interoperability with CUDA RT
+alignas(0x1000) thread_local volatile bool __dali_bench_force_tls_align;
+
+void __dali_bench_force_tls_align_fun(void) {
```

Comment:
I think in case of executables you have to actually call this function - and in case of shared objects, make it DLL_PUBLIC - with either of these, the variable itself doesn't need external linkage - to be kept alive, that is, I don't know what patchelf will do to it depending on visibility.